src/app/wallets/get-pending-onchain-balance-for-wallet.ts

```diff
@@ -24,3 +30,40 @@ export const getPendingOnChainBalanceForWallets = async (
 
   return IncomingOnChainTxHandler(pendingIncoming).balanceByWallet(wallets)
 }
+
+const briaGetPendingOnChainBalanceForWallets = async <S extends WalletCurrency>(
+  wallets: Wallet[],
+): Promise<{ [key: WalletId]: PaymentAmount<S> } | ApplicationError> => {
+  const pendingIncoming = await WalletOnChainPendingReceiveRepository().listByWalletIds({
+    walletIds: wallets.map((wallet) => wallet.id),
+  })
+  if (pendingIncoming instanceof CouldNotFindError) return {}
+  if (pendingIncoming instanceof Error) return pendingIncoming
+
+  const incomingTxHandler = NewIncomingOnChainTxHandler<S>(pendingIncoming)
+  if (incomingTxHandler instanceof Error) return incomingTxHandler
+
+  return incomingTxHandler.balanceByWallet(wallets)
+}
+
+export const getPendingOnChainBalanceForWallets = async <S extends WalletCurrency>(
+  wallets: Wallet[],
+): Promise<{ [key: WalletId]: PaymentAmount<S> } | ApplicationError> => {
+  const balancesFromLnd = await lndGetPendingOnChainBalanceForWallets(wallets)
+  if (balancesFromLnd instanceof Error) return balancesFromLnd
+
+  const balancesFromBria = await briaGetPendingOnChainBalanceForWallets<S>(wallets)
+  if (balancesFromBria instanceof Error) return balancesFromBria
+
+  const walletIds = Array.from(
+    new Set([...Object.keys(balancesFromLnd), ...Object.keys(balancesFromLnd)]),
+  ) as WalletId[]
```

Comment:
this confirms the issue of having two methods... we just need to replace/update getOnChainTxs (and persistence) with `WalletOnChainPendingReceiveRepository` and only have 1 method/ source of truth

Reply:
Resolved by passing everything thru WalleterPendingRepo